docs/REST API/rest-api.md

```diff
@@ -3826,6 +3828,102 @@ Retrieves a list of the configuration profiles assigned to a host.
 }
 ```
 
+### Live query one host (ad-hoc)
+
+Runs an ad-hoc live query against the specified host and responds with the results.
+
+The live query will stop if the targeted host is offline, or hasn't responded after 25 seconds (or whatever time period is configured via environment variable, e.g. `FLEET_LIVE_QUERY_REST_PERIOD=90s`).
+
+
+`POST /api/v1/fleet/hosts/:id/query`
+
+#### Parameters
+
+| Name      | Type  | In   | Description                                                                                                                                                        |
+|-----------|-------|------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| identifier       | integer  | path | **Required**. The target host ID. |
+| query            | string   | body | **Required**. The query SQL. |
+
+
+#### Example
+
+`POST /api/v1/fleet/hosts/123/query`
+
+##### Request body
+
+```json
+{
+  "query": "SELECT model, vendor FROM usb_devices;"
+}
+```
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "host_id": 123,
+  "query": "SELECT model, vendor FROM usb_devices;",
+  "status": "online", // "online" or "offline"
+  "error": null,
+  "rows": [
+    {
+      "model": "USB2.0 Hub",
+      "vendor": "VIA Labs, Inc."
+    }
+  ]
+}
+```
+
```

Comment:
```suggestion

Note that if the host is online and the query times out, this endpoint will return an error and `rows` will be `null`. If the host is offline, no error will be returned, and `rows` will be an empty array (`[]`).

```
@lucasmrod is this correct?

Reply:
One small amend:

> If the host is offline, no error will be returned, and `status` will be `offline` and `rows` will be `null`.

E.g.
```sh
curl -v -k -X POST -H "Authorization: Bearer $TEST_TOKEN" https://localhost:8080/api/latest/fleet/hosts/identifier/lucass-macbook-pro.local/query -d '{"query": "SELECT sleep(60);"}'
{
  "host_id": 3908,
  "rows": null,
  "query": "SELECT * from osquery_info;",
  "status": "offline"
}
```

Reply:
> Note that if the host is online and the query times out, this endpoint will return an error and `rows` will be `null`.

And when it times out the error is always `"error": "timeout waiting for results"`.